internal/command/jsonplan/values.go

```diff
@@ -199,17 +200,19 @@ func marshalPlanResources(changes *plans.Changes, ris []addrs.AbsResourceInstanc
 		// as the decoded value is only an intermediate step in transcoding
 		// this to a json format.
 		changeV.Before, _ = changeV.Before.UnmarkDeep()
-		changeV.After, _ = changeV.After.UnmarkDeep()
+		after, _ := changeV.After.UnmarkDeep()
```

Comment:
This is a bit uncomfortable to me - we need the marked value to identify sensitive attributes; it isn't clear if there is another reason these are unmarked into the same structure (changeV.After) instead of a new variable. I wouldn't have even thought to ask this if it weren't for the apparently unused, unmarked `changeV.Before` right above this.

Reply:
I did this specifically because we did _not_ want the marked values for the json output before now, and to make sure that we never attempted to use the marked value in the rest of the code which had no notion of "marked-ness". If we need both the marked and unmarked values, now is a great time to check all the references throughout `jsonplan`! 😄

Reply:
I don't have a good sense yet of what is and is not safe to do with marked values, so I instead copied the marked after values into a variable for use. Thanks @jbardin !